$OpenBSD: patch-make_autoconf_flags-cflags_m4,v 1.1 2019/07/23 11:55:42 kurt Exp $

On x86 (32bit) with clang add -mstackrealign to JVM CFLAGS:
* On x86 32bit, the JVM builds with -mstack-alignment=16 which is
  required for SSE instruction support in hotstpot. However, this option
  only maintains 16 byte alignment and does not realign the stack when
  called from native i386 code with 4 byte alignment. Jdk 8 had this
  option as well and was lost in the refactoring of the JDK in later
  versions.
* It also corrects fastdebug build assert failures in stubRoutines.cpp.

Index: make/autoconf/flags-cflags.m4
--- make/autoconf/flags-cflags.m4.orig
+++ make/autoconf/flags-cflags.m4
@@ -519,6 +519,10 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_HELPER],
     # works for all platforms.
     TOOLCHAIN_CFLAGS_JVM="$TOOLCHAIN_CFLAGS_JVM -mno-omit-leaf-frame-pointer -mstack-alignment=16"
 
+    if test "x$OPENJDK_TARGET_CPU" = xx86; then
+      TOOLCHAIN_CFLAGS_JVM="$TOOLCHAIN_CFLAGS_JVM -mstackrealign"
+    fi
+
     if test "x$OPENJDK_TARGET_OS" = xlinux; then
       TOOLCHAIN_CFLAGS_JDK="-pipe"
       TOOLCHAIN_CFLAGS_JDK_CONLY="-fno-strict-aliasing" # technically NOT for CXX
